src/delta/delta.c

```diff
@@ -207,102 +208,99 @@ static int found_override(const char *top, const char *bottom) {
         return k;
 }
 
-static int enumerate_dir_d(Hashmap *top, Hashmap *bottom, Hashmap *drops, const char *toppath, const char *drop) {
+static int enumerate_dir_d(Hashmap *bottom, const char *toppath, const char *drop) {
         _cleanup_free_ char *unit = NULL;
-        _cleanup_free_ char *path = NULL;
         _cleanup_strv_free_ char **list = NULL;
-        char **file;
-        char *c;
+        char **file, *c, *path;
         int r;
 
         assert(!endswith(drop, "/"));
 
-        path = strjoin(toppath, "/", drop, NULL);
-        if (!path)
-                return -ENOMEM;
+        log_debug("Looking at %s", toppath);
 
-        log_debug("Looking at %s", path);
-
-        unit = strdup(drop);
-        if (!unit)
+        path = strdup(toppath);
+        if (!path)
                 return -ENOMEM;
 
-        c = strrchr(unit, '.');
+        c = strrchr(path, '.');
         if (!c)
```

Comment:
```path``` is used as a key, I can't use ```_cleanup_free_```. Note that the original code was written in exactly the same manner.